script.js

```diff
@@ -1,4 +1,283 @@
+      "<br>" +
+      "🤞 You can decide the Rank of Ace to be either 1 or 11" +
+      "<br>" +
+      "👉 Type in your choice: 1 or 11 in the orange bar above."
+    );
+  }
+
+  if (gameMode == "ace1/11") {
+    gameMode = "dealCards";
+    var counter = 0;
+    while (counter < shuffledDeck.length) {
+      if (shuffledDeck[counter].name == "ace") {
+        shuffledDeck[counter].rank == input;
+      }
+      counter = counter + 1;
+    }
+    return (
+      "🃎 Ace will have Rank: " +
+      input +
+      " for this round" +
+      "<br>" +
+      "♣️♠️♦️❤️ So it's now the Blackjack time. 👍 Press the 'Submit' button now" +
+      "<br>" +
+      "🃏🃏 Dealer will shuffle the deck and pull out two cards for you."
+    );
+  }
+
+  if (gameMode == "dealCards") {
+    card1 = shuffledDeck.pop();
+    card2 = shuffledDeck.pop();
+    playerScore = card1.rank + card2.rank;
+    if (playerScore == 21) {
+      gameMode = "endGame/newGame";
+      decision1 =
+        "perfect Blackjack , congrats" + "<br>" + "select end/continue ?";
+    }
+    if (playerScore > 21) {
+      gameMode = "endGame/newGame";
+      decision1 = "Bust, you lost your bet" + "<br>" + "select end/continue ?";
+    }
+    if (playerScore < 21) {
+      gameMode = "hitStay";
+      decision1 =
+        ". you want 'hit' or 'stay' ?" +
+        "<br>" +
+        "Type in your decision 'hit' or 'stay' in the orange bar above.";
+    }
+
+    return (
+      playerName +
+      ", your Score is : " +
+      playerScore +
+      decision1 +
+      "<br>" +
+      "<br>" +
+      printCard(card1) +
+      printCard(card2)
+    );
+  }
+
+  if (gameMode == "hitStay") {
+    if (input == "hit") {
+      var card3 = shuffledDeck.pop();
+      playerScore = playerScore + card3.rank;
+
+      console.log(card3.rank, playerScore);
+
+      if (playerScore == 21) {
+        gameMode = "endGame/newGame";
+        decision2 =
+          ". Perfect Blackjack! you doubled your money to" +
+          yourBetAmount * 2 +
+          "<br>" +
+          "You want to 'end' or 'continue' the game? " +
+          "<br>" +
+          " Type in your decision 'end' or 'continue' in the orange bar above";
+      }
+      if (playerScore > 21) {
+        gameMode = "endGame/newGame";
+        decision2 =
+          ". Bust, you lost your bet amount" +
+          yourBetAmount +
+          "<br>" +
+          "You want to 'end' or 'continue' the game ? " +
+          "<br>" +
+          " Type in your decision 'end' or 'continue' in the orange bar above";
+      }
+      if (playerScore < 21) {
+        decision2 =
+          ". You want 'hit' or 'stay' ?" +
+          "<br>" +
+          "Type in your decision 'hit' or 'stay' in the orange bar above.";
+      }
+      return (
+        "Now your score is : " +
+        playerScore +
+        decision2 +
+        "<br>" +
+        "<br>" +
+        printCard(card3)
+      );
+    }
+
+    if (input == "stay") {
+      gameMode = "dealer";
```

Comment:
I discovered a bug when navigating through the game flow of your project. When the `gameMode` is set to `"dealer"`, the game breaks - the `main` function doesn't have a condition that checks if the `gameMode` is `"dealer"`. 

This resulted in a situation the `main` function does absolutely nothing, and returns `undefined` into your red output box. I have attached a screenshot below to show you what I meant.

<img width="1359" alt="Screenshot 2021-11-12 at 1 04 44 AM" src="https://user-images.githubusercontent.com/7672836/141339281-0e2a98d4-a047-4dc6-a53e-0fa8c22637c8.png">